### ISSUE TITLE
Detect when somebody seems to be being spammed with friend requests and offer to change the NoSpam

### ISSUE THREAD

**Issue by ghost** [qTox/qTox#2881]
I originally accidentally created the issue [here](https://github.com/irungentoo/toxcore/issues/1515#issuecomment-175785718) though I requested automatic NoSpam changing, but it was pointed out to me that this could cause a lot of confusion, so I'm going to go for what @Impyy:

> Maybe clients should implement detection for friend request spams and display a message that says: >"Tox has detected that you're being spammed with friend requests. Would you like to change your >nospam to counter this attack?".

**Comment by tux3** [comment]
Is this an actual problem that you or someone else is facing?
The NoSpam should deter friend request spam just by existing. I'm all for preventing spam and helping users discover features, but solving theoretical issues that don't actually affect anyone is not super high on my list of priorities :/

**Comment by zetok** [comment]
@tux3 

> Is this an actual problem that you or someone else is facing?

Since ~tox isn't really big, it's not a problem as of now.

Note that even if it doesn't seem like it, FRs also can be used by people for harassment. Imagine receiving very frequent FRs (that you reject) with some _really bad_ messages in them. This would be seriously disturbing for most of people.

e.g. qTox asking user whether they want to change NoSpam after they have declined FR from particular PK a few times (2 or 3) could be really nice from user's PoV.

I agree that it's not important enough to bother you with doing it - that's why label `help wanted`, which should signal that someone who wants to implement this particular feature should do it. :)

**Comment by LuccoJ** [comment]
I'd call this low priority, but it's an interesting idea when the overall goal is to make things user friendly without obscure options: the "nospam" thing isn't clear or intuitive, and this could be a way to "bring it to the masses" in a much more obvious way.

**Comment by tux3** [comment]
I agree that it's a good idea, and I'll be happy to merge that if someone wants to step up, but I need to prioritize in the meantime :)
